### PR TITLE
More f-strings mainly on sequence code, using flynt 0.69

### PR DESCRIPTION
**Pull request by peterjc**
- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``pre-commit`` locally,
and understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

Progress on #2510 using the latest version of ``flynt`` (default settings), which is more aggressive than ``pyupgrade`` and does not use the ``.format(...)`` method.

I have tried not to touch too many files in this PR, and have eyeballed all the changes. In a couple of places I made a manual tweak as well (e.g. combine string addition with percent formatting into a single f-string).

### CODE REVIEW

File: BioSQL/BioSeqDatabase.py [hunk 1]
```diff
@@ -185,7 +185,7 @@ def __init__(self, conn, module, module_name=None):
 
     def __repr__(self):
         """Return a short description of the class name and database connection."""
-        return self.__class__.__name__ + "(%r)" % self.adaptor.conn
+        return f"{self.__class__.__name__}({self.adaptor.conn!r})"
```
Comment (peterjc) [inline BioSQL/BioSeqDatabase.py:188]: This was manually edited to remove the string concatenation

**Comment by peterjc** [comment]
Sphinx is failing on CircleCI, looks like a coincidence with the jump from Sphinx v4.2.0 to Sphinx v4.2.1.

**Review by mdehoon — APPROVED** [review]

**Comment by peterjc** [comment]
Cross reference https://github.com/sphinx-doc/sphinx/issues/9631

**Comment by peterjc** [comment]
Sphinx issue avoided with 3f7a13d269ecb81011d7bf4f62c84fe50f626d33

**Pull request merged**